server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraWithTikaTest.java

```diff
@@ -19,28 +19,17 @@
 
 package org.apache.james;
 
-import java.io.IOException;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
-import org.junit.ClassRule;
-import org.junit.Rule;
+class CassandraWithTikaTest implements JamesServerContract {
 
-public class CassandraWithTikaTest extends AbstractJamesServerTest {
-
-    @ClassRule
-    public static final DockerCassandraRule cassandra = new DockerCassandraRule();
-    @ClassRule
-    public static final GuiceTikaRule guiceTikaRule = new GuiceTikaRule();
-
-    @Rule
-    public CassandraJmapTestRule cassandraJmap = CassandraJmapTestRule.defaultTestRule();
-
-    @Override
-    protected GuiceJamesServer createJamesServer() throws IOException {
-        return cassandraJmap.jmapServer(guiceTikaRule.getModule(), cassandra.getModule(), DOMAIN_LIST_CONFIGURATION_MODULE);
-    }
-
-    @Override
-    protected void clean() {
-    }
+    @RegisterExtension
+    static TikaExtension tikaExtension = new TikaExtension();
 
+    @RegisterExtension
+    static CassandraJmapTestExtension cassandraJmapServer = CassandraJmapTestExtension.builder()
+        .defaultCoreModule()
```

Comment:
I'd argue that ` .defaultCoreModule()` is compulsory, and do not demand any input from user.

Thus, as it is noise, you should not require me to call it.

Reply:
in the way I create my builder. Each method `defaultCoreModule()` or `defaultExtensions()`,... are not only for setting up default value but also for switching to the next `stage` of the builder. Hence it is compulsory. Like cascading lambdas but provide more options than just only one for each `stage`.

Traditional builder make the build implicit setup default value. I like the explicit one.

Reply:
I agree